pkg/flow/flow_http.go

```diff
@@ -49,6 +51,36 @@ func (f *Flow) ConfigHandler() http.HandlerFunc {
 	}
 }
 
+// ComponentHandler returns an http.HandlerFunc which will delegate all requests to
+// a component named by the first path segment
+func (f *Flow) ComponentHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		id := vars["id"]
+
+		// find node with ID
+		var node *controller.ComponentNode
+		for _, n := range f.loader.Components() {
+			if n.ID().String() == id {
+				node = n
+				break
+			}
+		}
+		if node == nil {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		handler := node.HTTPHandler()
+		if handler == nil {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		// remove /component/{id} from front of path, so each component can handle paths from their own root path
+		r.URL.Path = strings.TrimPrefix(r.URL.Path, "/component/"+id)
```

Comment:
Can we put a helper method in the `component` package to get a path prefix by component ID? It's being spread across two packages (here and the component) and it feels a little magic-stringy.

Reply:
Yes, I added HTTPPath to the options passed in at component creation time.